### DISCUSSION TITLE
v1.19.0

### DISCUSSION THREAD

**Discussion by sojan-official** [chatwoot/chatwoot#2834] [Releases & Announcements]
## Changelog

- Ability to snooze conversations
- Keyboard shortcuts
- Campaigns tab which now includes one-off SMS campaigns
- Email conversations now show subject, cc, and bcc
- Mobile app push notifications for self-hosted installations
- Docker image sizes are reduced by 50% ( special thanks to @paescuj )
- Chatwoot now blocks and throttle abusive requests
- Upgrade the ruby and rails version
- API for custom attributes
- API exposing widget configurations
- Numerous bug fixes and enhancements

Thanks to @SanjuSudheerM, @sbreiler-work, @paescuj, @raikasdev, @HariniKrishnan, @vishal-pandey, @shailesh-kalamkar, @patrickobriain  for the contributions

<hr /><em>This discussion was created from the release <a href='https://github.com/chatwoot/chatwoot/releases/tag/v1.19.0'>v1.19.0</a>.</em>

**Comment by Stephen-Lobo** [comment]
Thank you for new release and lot of new features.

I just tried to upgrade Chatwoot from 1.18 to the latest version, but it failed.

Ubuntu on Caprover DigitalOccean as per upgrade instructions at https://www.chatwoot.com/docs/self-hosted/deployment/caprover

We received following error.

`[Warning] One or more build-args [VAPID_PRIVATE_KEY POSTGRES_USERNAME REDIS_URL SMTP_ENABLE_STARTTLS_AUTO SMTP_PASSWORD MAILER_SENDER_EMAIL RAILS_INBOUND_EMAIL_PASSWORD MAILER_INBOUND_EMAIL_DOMAIN RAILS_LOG_TO_STDOUT ACTIVE_STORAGE_SERVICE SMTP_PORT POSTGRES_PASSWORD POSTGRES_HOST S3_BUCKET_NAME AWS_REGION IP_LOOKUP_SERVICE RAILS_INBOUND_EMAIL_SERVICE RAILS_ENV REDIS_PASSWORD SMTP_DOMAIN SMTP_USERNAME FRONTEND_URL POSTGRES_DATABASE AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY SMTP_ADDRESS SMTP_AUTHENTICATION SECRET_KEY_BASE IP_LOOKUP_API_KEY VAPID_PUBLIC_KEY] were not consumed
`

**Reply by pranavrajs** [reply]
@vishnu-narayanan Can you please take a look at this?

**Reply by vishnu-narayanan** [reply]
@Stephen-Lobo  Did the update fail ? Other than this warning, did it generate any errors ?

**Reply by Stephen-Lobo** [reply]
@vishnu-narayanan 

Pls see the attached Screenshot. I have tried three times so far, but same error.  It shows build successfully message. But, I dont see any new changes like Keyboard Shortcuts Or any other changes in the ChatWoot App to indicate 1.19 version (It would be awesome, if Chatwoot displays current installed version in App Profile page / configuration). So it looks upgrade is installing 1.18 files (not sure). 

If you can share your email address, I can share you credentials of Capover, so that you can install / debug yourself. 

https://user-images.githubusercontent.com/87435610/129885951-210d82e9-3226-4488-b546-9e641a2dc9fd.mov

**Reply by sojan-official** [reply]
@Stephen-Lobo  could you try using the specific version of the image instead, using the following snippets . seems like caprover is picking up cached images for latest.

# rails app (web)
```
{
  "schemaVersion": 2,
  "dockerfileLines": [
    "FROM chatwoot/chatwoot:v1.19.0",
    "RUN chmod +x docker/entrypoints/rails.sh",
    "ENTRYPOINT [\"docker/entrypoints/rails.sh\"]",
    "CMD bundle exec rake db:chatwoot_prepare; bundle exec rails s -b 0.0.0.0 -p 3000"
  ]
}
```
## sidekiq worker ( worker )
```
{
  "schemaVersion": 2,
  "dockerfileLines": [
    "FROM chatwoot/chatwoot:v1.19.0",
    "RUN chmod +x docker/entrypoints/rails.sh",
    "ENTRYPOINT [\"docker/entrypoints/rails.sh\"]",
    "CMD bundle exec sidekiq -C config/sidekiq.yml"
  ]
}
```

to know the version of chatwoot you have installed you can hit the url `yourinstallationurl/api ` or you go go into the account settings pages. 

<img width="1296" alt="Screenshot 2021-08-18 at 4 56 18 PM" src="https://user-images.githubusercontent.com/73185/129890183-d1af4697-52af-4bae-856e-2b5a1536f31b.png">

**Reply by Stephen-Lobo** [reply]
Thank you for prompt reply. I just tried, but received this message :

`Wed Aug 18 2021 11:38:15 GMT+0000 (Coordinated Universal Time)
Build started for chatwoot-web
Ignore warnings for unconsumed build-args if there is any
Step 1/4 : FROM chatwoot/chatwoot:v1.19.0

{"message":"Get https://registry-1.docker.io/v2/: net/http: request canceled (Client.Timeout exceeded while awaiting headers)"}
Get https://registry-1.docker.io/v2/: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
Build has failed!
----------------------
Deploy failed!
Error: {"message":"Get https://registry-1.docker.io/v2/: net/http: request canceled (Client.Timeout exceeded while awaiting headers)"}
Get https://registry-1.docker.io/v2/: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
`

**Reply by Stephen-Lobo** [reply]
New Update:
I could successfully upgrade from version 1.18.0 to 1.18.1 using following method. But when I attempted to upgrade to 1.19.0 by putting v1.19.0, it again failed with same message mentioned in earlier post.  

`{
  "schemaVersion": 2,
  "dockerfileLines": [
    "FROM chatwoot/chatwoot:v1.18.1",
    "RUN chmod +x docker/entrypoints/rails.sh",
    "ENTRYPOINT [\"docker/entrypoints/rails.sh\"]",
    "CMD bundle exec rake db:chatwoot_prepare; bundle exec rails s -b 0.0.0.0 -p 3000"
  ]
}`

**Reply by Stephen-Lobo** [reply]
New Update:

We tried once again to install 1.19v. Got successful Message :

```
Thu Aug 19 2021 11:29:40 GMT+0000 (Coordinated Universal Time)
Build started for chatwoot-worker
Ignore warnings for unconsumed build-args if there is any
Step 1/4 : FROM chatwoot/chatwoot:v1.19.0

---> 20176e93402e
Step 2/4 : RUN chmod +x docker/entrypoints/rails.sh

---> Using cache
---> c46ba10c7f6e
Step 3/4 : ENTRYPOINT ["docker/entrypoints/rails.sh"]

---> Using cache
---> c65d65d338ac
Step 4/4 : CMD bundle exec sidekiq -C config/sidekiq.yml

---> Using cache
---> c496addd848d
[Warning] One or more build-args [POSTGRES_DATABASE REDIS_PASSWORD VAPID_PUBLIC_KEY POSTGRES_HOST REDIS_URL IP_LOOKUP_API_KEY FRONTEND_URL VAPID_PRIVATE_KEY AWS_ACCESS_KEY_ID SMTP_AUTHENTICATION RAILS_INBOUND_EMAIL_SERVICE RAILS_INBOUND_EMAIL_PASSWORD MAILER_INBOUND_EMAIL_DOMAIN SECRET_KEY_BASE AWS_REGION POSTGRES_PASSWORD RAILS_LOG_TO_STDOUT SMTP_ADDRESS SMTP_USERNAME RAILS_ENV POSTGRES_USERNAME SMTP_DOMAIN SMTP_ENABLE_STARTTLS_AUTO SMTP_PASSWORD ACTIVE_STORAGE_SERVICE S3_BUCKET_NAME SMTP_PORT MAILER_SENDER_EMAIL AWS_SECRET_ACCESS_KEY IP_LOOKUP_SERVICE] were not consumed
Successfully built c496addd848d
Successfully tagged img-captain-chatwoot-worker:latest
Build has finished successfully!
```

But, now at the frontend, We're getting :

```
NGINX 502 Error :/
If you are the developer, check your application's logs. 
```

<img width="639" alt="Screenshot 2021-08-19 at 5 01 51 PM" src="https://user-images.githubusercontent.com/87435610/130062707-11da7db0-c076-4e3a-a0cd-3f3cf2dc1869.png">

We have already re-booted docker as well as droplet. It didnt work.

@vishnu-narayanan  @sojan-official 

Can you please help us in upgrade. We have been waiting for 'Mobile push notifications' since months which is available in latest upgrade. Thank you in advance.

**Reply by vishnu-narayanan** [reply]
@Stephen-Lobo Apologies for the inconvenience. We are debugging this on our end and will keep you posted. 
https://github.com/chatwoot/chatwoot/issues/2843

**Reply by vishnu-narayanan** [reply]
@Stephen-Lobo This is due to an upstream issue with `docker` and `alpine` Linux on which chatwoot is based. We are working on a fix from our end.

Ref: https://github.com/chatwoot/chatwoot/issues/2845

**Reply by Stephen-Lobo** [reply]
I am glad to know that this issue is confirmed as a bug due to ruby-alpine-docker conflict & you are working on it.

BTW, you guys are terrific. The way Chatwoot Team, listens their customers / product-users & tries to help them, I am sure ChatWoot is just few years to become another successful Unicorn startup like PostMan.com

**Reply by vishnu-narayanan** [reply]
Thank you for your kind words. We are humbled. 🙇

**Reply by vishnu-narayanan** [reply]
Check out this comment to get it working in the interim.
https://github.com/chatwoot/chatwoot/issues/2843#issuecomment-902674022

**Reply by Stephen-Lobo** [reply]
Thank you. 

The hot-fix version has fixed the issue.

**Comment by helpobras** [comment]
my update is OK. but I found something that appears to be a glitch. an agent or an administrator user is able to create another account freely on mine where the username is.
I tested it with an admin user and a common agent

**Reply by sojan-official** [reply]
Did you mean that other people can sign up? 
if so please check the value of `ENABLE_ACCOUNT_SIGNUP` env variable

**Reply by helpobras** [reply]
Is not it. I'm saying that a user agent can create another account after logged in.
look at the video. https://youtu.be/cRn0h9BKS-Q
the ENABLE_ACCOUNT_SIGNUP parameter is api_only
I believe that the agent could not have this type of permission...

**Reply by sojan-official** [reply]
Did you change any of the installation configs? 
In production environments, Chatwoot has installation config  `CREATE_NEW_ACCOUNT_FROM_DASHBOARD` set as false. If you change that to true, the create button will appear.

**Reply by helpobras** [reply]
the only modification I made was to the .env file
ENABLE_ACCOUNT_SIGNUP=api_only

**Reply by helpobras** [reply]
I didn't find this parameter in the setup.sh file or in the .env file:
CREATE_NEW_ACCOUNT_FROM_DASHBOARD

**Comment by Stephen-Lobo** [comment]
Changelog of v1.19.0 mentions:

> Mobile app push notifications for self-hosted installations

But, after installing the latest version, we are not getting any notifications of new conversation on Mobile.

Please clarify.

**Reply by sojan-official** [reply]
Could you reach out on `hello@chatwoot.com`. We will have one of our engineers take a look at the installation.
